### ISSUE TITLE
win.exe package for RH 7

### ISSUE THREAD

**Issue by dmurphy18** [saltstack/salt#30658]
see win.exe package for RH 6 #21256

Ran out time with support for RH 7 when working on #21256, this issue is ensure the completion of winexe support on RH7 does not get dropped

**Comment by beardedeagle** [comment]
@dmurphy18: ooook, the other issue is closed so I will comment here.

```
[root@[REDACTED] ~]# cat /etc/redhat-release
CentOS Linux release 7.2.1511 (Core)
[root@[REDACTED] ~]# winexe --version
winexe version 1.1
This program may be freely redistributed under the terms of the GNU GPLv3
[root@[REDACTED] ~]# yum info winexe
Installed Packages
Name        : winexe
Arch        : x86_64
Version     : 1.1
Release     : b787d2.el7.centos
Size        : 7.8 M
Repo        : installed
Summary     : Remote Windows command executor.
URL         : http://sourceforge.net/projects/winexe/
License     : GPLv3
Description : Winexe remotely executes commands on Windows
            : NT/2000/XP/2003/Vista/7/2008/8/2012 systems from GNU/Linux.
[root@[REDACTED] ~]# winexe -U '[REDACTED]%[REDACTED]' //[REDACTED] "ipconfig"

Windows IP Configuration


Ethernet adapter floating ips:

   Connection-specific DNS Suffix  . :
   Link-local IPv6 Address . . . . . : [REDACTED]
   Autoconfiguration IPv4 Address. . : [REDACTED]
   Subnet Mask . . . . . . . . . . . : 255.255.0.0
   Default Gateway . . . . . . . . . :

Ethernet adapter Ethernet:

   Connection-specific DNS Suffix  . :
   Link-local IPv6 Address . . . . . : [REDACTED]
   IPv4 Address. . . . . . . . . . . : [REDACTED]
   Subnet Mask . . . . . . . . . . . : 255.255.252.0
   Default Gateway . . . . . . . . . : [REDACTED]

Tunnel adapter isatap.{6B31DE02-4AF0-4936-8E77-03FA8A03846F}:

   Media State . . . . . . . . . . . : Media disconnected
   Connection-specific DNS Suffix  . :

Tunnel adapter isatap.{1AB8D79F-7466-4263-8460-E05F1F980950}:

   Media State . . . . . . . . . . . : Media disconnected
   Connection-specific DNS Suffix  . :
```

Just doing some cleanup and then getting it commited to git. May or maynot try to dive into how to make it worthy of salt-pack. Idk, never dealt with that.

**Comment by beardedeagle** [comment]
I will also make mention that at my job I cannot install the winexesvc successfully until I have ran the following:

`reg ADD "HKLM\Software\Microsoft\Windows\CurrentVersion\Policies\system" /v LocalAccountTokenFilterPolicy /t REG_DWORD /d 1 /f
`

This along with ensuring that the `win_user` is in the administrators group solves those pesky error 241 issues.

Here is my full user_data file for provisioning via cloudbase-init (I use old school cmd, tends to be magnitudes faster as well as lighter on resources):

```
#ps1_sysnative

# add LocalAccountTokenFilterPolicy to registry
reg ADD "HKLM\Software\Microsoft\Windows\CurrentVersion\Policies\system" /v LocalAccountTokenFilterPolicy /t REG_DWORD /d 1 /f

# create winexe firewall rule and configure winexe service for port 445
netsh advfirewall firewall add rule name="winexe445" dir=in action=allow protocol=TCP localport=445
winrm set winrm/config/Listener?Address=*+Transport=HTTP @{Port="445"}
net stop WinRM
net Start WinRM

#Disable the OpenStack Scheduled Task to Update Local Administrators
schtasks /run /TN "Update Local Admins"
schtasks /change /TN "Update Local Admins" /DISABLE

#Create [USERHERE] Local Account and Set Password
net user /add [USERHERE] [PASSWORDHERE]

#Add [USERHERE] to Local Administrators
net localgroup "administrators" [USERHERE] /add
```

**Comment by dmurphy18** [comment]
@beardedeagle I can help and/or do, with getting it into salt-pack.  Just let me know when satisfied and ready to try.  I have a bit of a back-log to get through of other stuff, but I think it should not take more than a half-day to get it into salt-pack.

**Comment by beardedeagle** [comment]
Sure, doing a quick rewrite to add backwards compatibility so it can be used against centos 6.* and centos 7.*

**Comment by beardedeagle** [comment]
@dmurphy18: ran into a speed bump. The source `.tar.gz` is 296mb. This is required because the full source of samba is needed to build winexe 1.1 to work with centos 7. Does salt have git lfs? if not I will need to modify the init.sls to just pull the sources down.

**Comment by dmurphy18** [comment]
@beardedeagle Checked with terminalmage and Saltstack does not support it yet.  On the list to do, but lower priority at the moment.

Note: just created a 'develop' branch for salt-pack.  All PR's should be against that branch in the future.
Documentation update is in progress.

**Comment by beardedeagle** [comment]
Looks like I will need to pull the sources then, rather than provide them. I'll get to it.

**Comment by beardedeagle** [comment]
I believe I have a solution, testing today.

**Comment by dmurphy18** [comment]
@beardedeagle I have a fix for the winexe-1.1.tar.gz to allow for Amazon Linux.  How do you want to handle it, you can contact me at SaltStack directly if you don't want to use this issue
